packages/server/lib/plugins/util.js

```diff
@@ -4,11 +4,62 @@ const debug = require('debug')('cypress:server:plugins')
 const Promise = require('bluebird')
 
 const UNDEFINED_SERIALIZED = '__cypress_undefined__'
+const FUNCTION_SERIALIZED = '__cypress_function__'
 
 const serializeError = (err) => {
   return _.pick(err, 'name', 'message', 'stack', 'code', 'annotated', 'type')
 }
 
+function serializeArgument (arg) {
+  if (arg === null || arg === undefined) {
+    return null
+  }
+
+  if (typeof arg === 'function') {
+    return FUNCTION_SERIALIZED
+  }
+
+  if (typeof arg === 'object') {
+    if (_.isArray(arg)) {
+      return arg.map((argElement) => serializeArgument(argElement))
+    }
+
+    return Object.keys(arg).reduce(function (acc, key) {
+      acc[key] = serializeArgument(arg[key])
+
+      return acc
+    }, {})
```

Comment:
I agree that this would be less verbose, but it has 2 issue in my opinion:
- It is a lot more obscure than what is there right now
- It creates a new object at every iteration

I will try the Object.entries approach. I think it's a good one.

Reply:
```js
const serializedObject = {}

    for (const [key, val] of Object.entries(arg)) {
      serializedObject[key] = serializeArgument(val)
    }

    return serializedObject
```

Reply:
Done in 00c799d5f03467ce535043e459acd72c6d89ebab